### ISSUE TITLE
`array-element-newline`: change default from "always" to "{ consistent: true, multiline: true }"

### ISSUE THREAD

**Issue by vladshcherbin** [eslint-stylistic/eslint-stylistic#1262]
### Clear and concise description of the problem

`array-element-newline` defaults to `"always"`, so the bare rule flags any multi-element array written on one line:

```js
const pair = [1, 2, 3]
// error: There should be a linebreak after this element. (x2)
```

Nobody formats a three-number array as a column. Enabling the rule without options punishes the layout that nearly all code uses, so in practice the first thing [everyone does](https://github.com/search?q=%22%40stylistic%2Farray-element-newline%22&type=code) is override the default.

### Suggested solution

`"consistent"` would already be a sane default. Both normal layouts pass, all elements on one line or one per line, and the rule keeps flagging the mixed case it exists for:

```js
const mixed = [1,
  2, 3]
// error
```

But I would go with `{ consistent: true, multiline: true }`, which #445 made possible. Plain `"consistent"` only looks at the breaks between elements, so it accepts this:

```js
const items = [{
  a: 1
}, { b: 2 }]
// no error with "consistent"
```

With `multiline: true` on top, one multiline element forces every element onto its own line:

```js
const items = [
  {
    a: 1
  },
  { b: 2 }
]
```

That is the layout people write for arrays of multiline objects anyway, and the single-line and column cases stay untouched.

### Additional context

v6 is in beta. A default change fits a major, and if it misses this one it waits a full cycle.

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by 9romise** [comment]
Thanks for the proposal. We’re going to close this PR because the affected rule is being superseded by `@stylistic/list-style`, which is now stable in v6.
The legacy rule will be deprecated and behaviorally frozen, so changing its default would add a new behavior change to a rule users are being asked to migrate away from. If the desired behavior is still needed, please propose it as an improvement to `list-style` instead.
